### PR TITLE
[DPE-4977] Run integration tests against Juju 3.6

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
We need to soon support this charm on juju 3.6. In order to prepare, we are going to run the integration tests against juju 3.6 on a nightly schedule

## Solution
First run juju 3.6 tests on PR to ensure that code is geared towards successful runs
Then use dpw to run juju 3.6 tests only on a nightly schedule

### CODE REVIEW

**Comment by shayancanonical** [comment]
[The run for 7107c2a](https://github.com/canonical/postgresql-k8s-operator/actions/runs/11729797288/job/32890747773?pr=689) shows that all tests (except `test_backups.py` on GCP) pass on either arm64 or amd64. As a result, we should run juju 3.6 tests on a nightly schedule and attempt to iteratively debug the issues preventing from fully green pipelines.

File: tests/integration/ha_tests/test_rollback_to_master_label.py
Comment (dragomirp) [inline tests/integration/ha_tests/test_rollback_to_master_label.py:1]: Marker on line 39 should be skipping this test for juju 2. We should investigate why the CI stopped doing that (ci.yaml changes?).

File: tests/integration/ha_tests/test_upgrade_to_primary_label.py [hunk 1]
```diff
@@ -42,13 +42,14 @@ async def test_deploy_stable(ops_test: OpsTest) -> None:
             num_units=3,
             channel="14/stable",
             revision=(280 if architecture == "arm64" else 281),
-            series=CHARM_SERIES,
             trust=True,
+            base=CHARM_BASE,
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/ha_tests/test_upgrade_to_primary_label.py]: this suggestion worked wonderfully. updated in [2e95476](https://github.com/canonical/postgresql-k8s-operator/pull/689/commits/2e95476cd405dac3bf119bd1939f0edee6649c80)

File: tests/integration/new_relations/test_new_relations.py [hunk 2]
```diff
@@ -705,12 +707,17 @@ async def test_indico_datatabase(ops_test: OpsTest) -> None:
     async with ops_test.fast_forward(fast_interval="30s"):
         await ops_test.model.deploy(
             "indico",
-            channel="stable",
+            channel="latest/edge",
             application_name="indico",
             num_units=1,
+            base=CHARM_BASE,
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/new_relations/test_new_relations.py]: Had to revert to using `series` in [8ad2403](https://github.com/canonical/postgresql-k8s-operator/pull/689/commits/8ad240371d415deca9723b31590d082304c0efa6) as we were only able to set the base as `ubuntu@22.04` 

error when using `ubuntu@22.04` base:
```
juju.errors.JujuError: base "ubuntu@20.04/stable" is not supported, supported bases are: ubuntu@22.04
```

**Review by dragomirp — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**